pmd-java/src/main/java/net/sourceforge/pmd/lang/java/multifile/signature/JavaOperationSignature.java

```diff
@@ -29,7 +30,7 @@
  */
 public final class JavaOperationSignature extends JavaSignature<ASTMethodOrConstructorDeclaration> {
 
-    private static final Map<Integer, JavaOperationSignature> POOL = new HashMap<>();
+    private static final Map<Integer, JavaOperationSignature> POOL = new ConcurrentHashMap<>();
```

Comment:
If the crash was caused by resize operations, maybe we can fill in the initial capacity (32) too to eliminate them. Granted, it probably won't change much since the current capacity is the default of 16.

Reply:
initial capacity indicates the initial number of buckets, but we can't guarantee the hash function will place each entry on a separate bucket, so resizes can still happen. We should just err on being safe.

Reply:
You're right of course